drivers/sensor/fxas21002/fxas21002.c

```diff
@@ -277,23 +279,25 @@ static const struct sensor_driver_api fxas21002_driver_api = {
 #endif
 };
 
-static const struct fxas21002_config fxas21002_config = {
-	.i2c = I2C_DT_SPEC_INST_GET(0),
-	.whoami = CONFIG_FXAS21002_WHOAMI,
-	.range = CONFIG_FXAS21002_RANGE,
-	.dr = CONFIG_FXAS21002_DR,
-#ifdef CONFIG_FXAS21002_TRIGGER
-#ifdef CONFIG_FXAS21002_DRDY_INT1
-	.int_gpio = GPIO_DT_SPEC_INST_GET(0, int1_gpios),
-#else
-	.int_gpio = GPIO_DT_SPEC_INST_GET(0, int2_gpios),
-#endif
-#endif
-};
-
-static struct fxas21002_data fxas21002_data;
-
-DEVICE_DT_INST_DEFINE(0, fxas21002_init, NULL,
-		    &fxas21002_data, &fxas21002_config,
-		    POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,
-		    &fxas21002_driver_api);
+#define FXAS21002_DEFINE(inst)									\
+	static struct fxas21002_data fxas21002_data_##inst;					\
+												\
+	static const struct fxas21002_config fxas21002_config_##inst = {			\
+		.i2c = I2C_DT_SPEC_INST_GET(inst),						\
+		.whoami = CONFIG_FXAS21002_WHOAMI,						\
+		.range = CONFIG_FXAS21002_RANGE,						\
+		.dr = CONFIG_FXAS21002_DR,							\
+		IF_ENABLED(CONFIG_FXAS21002_TRIGGER,						\
+			   (COND_CODE_1(CONFIG_FXAS21002_DRDY_INT1,				\
```

Comment:
I think the right thing to do would be to make choice based on the existence of `int1` or `int2` props. Kconfig options apply to all instances, which isn't great.

Reply:
Doesn't that assume that only one of them will be present in the device tree at any time?

Shouldn't we come up with a way that allows both to be present and some way of selecting which one is used?

Reply:
Looking at the hexiwear_k64 board and it has both `int1-gpios` and `int2-gpios` present. Checking for existence would require the user to comment/uncomment lines in the device tree to move between the pins, which I feel is quite undesirable.

Reply:
Would it make sense to use <specifier>-map for this?

Reply:
Any idea of how we could specify which one is used? I'm thinking of making another passage through the sensor drivers to move configurations to the device tree in order to have true instance configurability. However, there's lots of questions on my side on how to accomplish this.

Reply:
see for example lsm6dso